colour/algebra/interpolation.py

```diff
@@ -511,42 +1100,358 @@ class PchipInterpolator(scipy.interpolate.PchipInterpolator):
     Interpolates a 1-D function using Piecewise Cubic Hermite Interpolating
     Polynomial interpolation.
 
+    Attributes
+    ----------
+    y
+
     Notes
     -----
-    This class is a wrapper around *scipy.interpolate.PchipInterpolator*
-    class.
+    -   This class is a wrapper around *scipy.interpolate.PchipInterpolator*
+        class.
     """
 
-    def __init__(self, x=None, y=None, *args, **kwargs):
+    def __init__(self, x, y, *args, **kwargs):
         super(PchipInterpolator, self).__init__(x, y, *args, **kwargs)
 
         self._y = y
 
     @property
     def y(self):
         """
-        Property for **self._y** private attribute.
+        Getter and setter property for the dependent and already known
+        :math:`y` variable.
+
+        Parameters
+        ----------
+        value : array_like
+            Value to set the dependent and already known :math:`y` variable
+            with.
 
         Returns
         -------
         array_like
-            self._y
+            Dependent and already known :math:`y` variable.
         """
 
         return self._y
 
     @y.setter
     def y(self, value):
         """
-        Setter for **self._y** private attribute.
+        Setter for the **self.y** property.
+        """
+
+        raise AttributeError('"{0}" attribute is read only!'.format('y'))
+
+
+class NullInterpolator(object):
+    """
+    Performs 1-D function null interpolation, i.e. a call within given
+    tolerances will return existing :math:`y` variable values and `default` if
+    outside tolerances.
+
+    Parameters
+    ----------
+    x : ndarray
+        Independent :math:`x` variable values corresponding with :math:`y`
+        variable.
+    y : ndarray
+        Dependent and already known :math:`y` variable values to
+        interpolate.
+    absolute_tolerance : numeric, optional
+        Absolute tolerance.
+    relative_tolerance : numeric, optional
+        Relative tolerance.
+    default : numeric, optional
```

Comment:
If it's optional, what is the default `default`? (This might be important for all other optional parameters)

Reply:
Or is this information automatically included in the docs from the signature?

Reply:
The built documentation includes all the default values in the definition signature, e.g.: https://colour.readthedocs.io/en/latest/colour.algebra.interpolation.html#colour.algebra.interpolation.lagrange_coefficients

I find it to be a bit redundant to put defaults there (and they must be maintained too).

Reply:
Yes, agree in that case there's no benefit to that.